pyphare/pyphare/pharesee/hierarchy/tensorfield.py

```
@@ -0,0 +1,97 @@
+def GetDomainSize(hier, **kwargs):
+    root_cell_width = hier.level(0).cell_width
+    domain_box = hier.domain_box
+    return (domain_box.upper + 1) * root_cell_width
+
+
+def GetDl(hier, time, level="finest"):
+    level = hier.finest_level(time) if level == "finest" else level
+    return hier.level(level).cell_width
+
+
+def get_interpolated_selection_from(hier: AnyTensorField, input, interp="nearest"):
```

Ruff (0.15.9)

[error] Function argument `input` is shadowing a Python builtin

(A002)